kubetools/kubernetes/config/__init__.py

```diff
@@ -353,6 +353,8 @@ def generate_kubernetes_configs_for_project(
         schedule = cronjob['schedule']
         concurrency_policy = cronjob['concurrency_policy']
         batch_api_version = cronjob.get('batch-api-version')  # May depend on target cluster
+        successfulJobsHistoryLimit = cronjob.get('successfulJobsHistoryLimit', 3)
+        failedJobsHistoryLimit = cronjob.get('failedJobsHistoryLimit', 1)
```

Comment:
This is now at least 5 fields that are directly copied from `kubetools.yml` to the cronjob k8s definition. It's time to make this generic so that we don't need a PR like this one each time.

Reply:
Is this what you meant ?
https://github.com/EDITD/kubetools/pull/166/commits/917151178ebcea9e1e51ee867daa97a46af0a2f9

Reply:
We could do something like

```
def _additional_cronjob_params(cronjob, cronjob_spec):
    for param, value in cronjob.items():
      cronjob_spec[param] = value
    return cronjob_spec

def _required_cronjob_params(cronjob, cronjob_spec):
  required_params = ['schedule', 'concurrency_policy']
  for param in required_params:
    # might want to consider how we fail here if it's not present
    cronjob_spec[param] = cronjob.pop(param)
  return cronjob, cronjob_spec
  
# Back to main function
....
  cronjob_spec = {}
  cronjob, cronjob_spec = _required_cronjob_params(cronjob, cronjob_spec)
  cronjob_spec = _additional_cronjob_params(cronjob, cronjob_spec)
....
```

Reply:
https://github.com/EDITD/kubetools/pull/166/commits/dd9a16f9d019af83d9fa2f7fad86712a190e7063